components/config/opts.rs

```diff
@@ -667,13 +666,12 @@ pub fn from_cmdline_args(args: &[String]) -> ArgumentParsingResult {
 
     let url = match url_opt {
         Some(url_string) => {
-            parse_url_or_filename(&cwd, url_string)
-                .unwrap_or_else(|()| args_fail("URL parsing failed"))
+            match parse_url_or_filename(&cwd, url_string) {
+                Ok(v) => Some(v),
+                Err(_) => None,
+            }
```

Comment:
You can transform an Result into an Option with parse_url_or_filename().ok().
See https://doc.rust-lang.org/std/result/enum.Result.html#method.ok